packages/vue/src/components/IonIcon.ts

```diff
@@ -19,11 +20,15 @@ export const IonIcon = /*@__PURE__*/ defineComponent({
   setup(props, { slots }) {
     defineCustomElement();
     return () => {
-      const { icon, ios, md } = props;
+      const { icon, ios, md, mode } = props;
 
       let iconToUse: typeof icon;
+
+      const config = getConfig();
```

Comment:
`getConfig` can return null, so we should have a fallback for when that is the case: https://github.com/ionic-team/ionic-framework/blob/27a5356fa2b72073d565e9d6f527107869faa3ee/packages/vue/src/utils.ts#L66